runtime/sam/expr/coerce/coerce.go

```diff
@@ -17,7 +19,13 @@ func Equal(a, b super.Value) bool {
 	}
 	switch aid, bid := a.Type().ID(), b.Type().ID(); {
 	case !super.IsNumber(aid) || !super.IsNumber(bid):
-		return aid == bid && bytes.Equal(a.Bytes(), b.Bytes())
+		if aid != bid {
+			return false
+		}
+		if aid == super.IDNet {
+			return NetCompare(super.DecodeNet(a.Bytes()), super.DecodeNet(b.Bytes())) == 0
+		}
```

Comment:
I don't think we need this. Tests pass without it.

Reply:
I don't like have the compare be different than what we do on the vector side. Maybe it all works out fine, maybe it doesn't our testing isn't the most robust in this area. If you want I can change vector side to convert nets to bytes and compare that?

Reply:
Then my vote's for using `==` on netip.Prefixes in both runtimes.